webapp/lib/app/configurator/standard_messages/controller_messages_helper.dart

```diff
@@ -9,81 +22,184 @@ class StandardMessagesManager {
-    List<model.SuggestedReply> updated = [];
-    for (var standardMessage in standardMessages) {
-      var result = updateStandardMessage(standardMessage);
-      if (result != null) updated.add(result);
+  model.SuggestedReply onUpdateStandardMessageFromStorage(model.SuggestedReply standardMessage) {
+    storageCategories.putIfAbsent(standardMessage.categoryId, () => MessageCategory(standardMessage.categoryId, standardMessage.categoryName, standardMessage.categoryIndex));
+    storageCategories[standardMessage.categoryId].groups.putIfAbsent(standardMessage.groupId, () => MessageGroup(standardMessage.groupId, standardMessage.groupName, standardMessage.groupIndexInCategory));
+    storageCategories[standardMessage.categoryId].groups[standardMessage.groupId].messages.putIfAbsent(standardMessage.suggestedReplyId, () => standardMessage.clone());
+    storageCategories[standardMessage.categoryId].groups[standardMessage.groupId].messages[standardMessage.suggestedReplyId] = standardMessage.clone();
+    return standardMessage;
+  }
+
+  List<model.SuggestedReply> onUpdateStandardMessagesFromStorage(List<model.SuggestedReply> standardMessagesToAdd) {
+    List<model.SuggestedReply> added = [];
+    for (var standardMessage in standardMessagesToAdd) {
+      onUpdateStandardMessageFromStorage(standardMessage);
+
+      // todo: check if unedited
+      var result = updateStandardMessageInLocal(standardMessage);
+      if (result != null) added.add(result);
     }
-    return updated;
+    return added;
   }
 
-  model.SuggestedReply removeStandardMessage(model.SuggestedReply standardMessage) {
-    if (!standardMessages.any((element) => element.suggestedReplyId == standardMessage.suggestedReplyId)) {
+  model.SuggestedReply removeStandardMessageInLocal(model.SuggestedReply standardMessage) {
+    // todo: this might not be relevant
+    if (!standardMessagesInLocal.any((element) => element.suggestedReplyId == standardMessage.suggestedReplyId)) {
       log.warning("Standard messages consistency error: Removing message that doesn't exist: ${standardMessage.suggestedReplyId}");
       return null;
     }
-    var oldStandardMessage = standardMessages.singleWhere((element) => element.suggestedReplyId == standardMessage.suggestedReplyId);
-    categories[oldStandardMessage.categoryId].groups[oldStandardMessage.groupId].messages.remove(oldStandardMessage.suggestedReplyId);
+    var oldStandardMessage = standardMessagesInLocal.singleWhere((element) => element.suggestedReplyId == standardMessage.suggestedReplyId);
+    localCategories[oldStandardMessage.categoryId].groups[oldStandardMessage.groupId].messages.remove(oldStandardMessage.suggestedReplyId);
     return standardMessage;
   }
 
-  List<model.SuggestedReply> removeStandardMessages(List<model.SuggestedReply> standardMessages) {
+  model.SuggestedReply onRemoveStandardMessageFromStorage(model.SuggestedReply standardMessage) {
```

Comment:
nit - let's rename this slightly and make it a private method: `_removeStandardMessageInStorage`